src/transformers/tokenization_utils.py

```diff
@@ -49,6 +49,126 @@
 TOKENIZER_CONFIG_FILE = "tokenizer_config.json"
 
 
+class Trie:
+    """
+    Trie in Python. Creates a Trie out of a list of words. The trie is used to split on `added_tokens` in one pass
```

Comment:
Link to the algorithm for those who don't know it? -> https://en.wikipedia.org/wiki/Trie